raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -1814,6 +1814,84 @@ def test_filter_reachable_routes():
     assert possible_routes[1] not in filtered_routes
 
 
+def test_resume_waiting_transfer():
+    """ Test that a mediator who has a waiting_transfer
+    set (the transfer couldn't be sent forward or backward
+    due to availability or capacity or timeout issues) will retry
+    mediating the waiting_transfer as soon as this transfer's
+    lock expiration becomes valid.
+    """
+    setup = factories.make_transfers_pair(2)
+
+    # Also add transfer sender channel
+    partner_state = factories.NettingChannelEndStateProperties(address=UNIT_TRANSFER_SENDER)
+    payer_channel = factories.create(
+        factories.NettingChannelStateProperties(partner_state=partner_state)
+    )
+    setup.channels.channels.append(payer_channel)
+
+    possible_routes = setup.channels.get_routes()
+
+    # we received a transfer that expires after settle timeout would hit
+    lock_expiration = UNIT_SETTLE_TIMEOUT + 3
+    received_transfer = factories.create(
+        factories.LockedTransferSignedStateProperties(
+            amount=1,
+            expiration=lock_expiration,
+            canonical_identifier=payer_channel.canonical_identifier,
+        )
+    )
+
+    # we simulate that the initial transfer came to early (from our point of view)
+    # so it went into `waiting_transfer` state
+    # this can happen, when reveal_timeout is around `settle_timeout / 2` and the
+    # receiving node is slightly behind with the blockchain sync
+    # see: https://github.com/raiden-network/raiden/issues/4998
+    mediator_state = MediatorTransferState(secrethash=UNIT_SECRETHASH, routes=possible_routes)
+    mediator_state.waiting_transfer = WaitingTransferState(received_transfer)
```

Comment:
Please replace this with the actual state change state transition and afterwards assert that the mediator is setting `waiting_transfer`.